core/src/main/java/io/substrait/relation/RelCopyOnWriteVisitor.java

```diff
@@ -342,6 +342,26 @@ public Optional<Rel> visit(NestedLoopJoin nestedLoopJoin) throws EXCEPTION {
             .build());
   }
 
+  @Override
+  public Optional<Rel> visit(ConsistentPartitionWindow consistentPartitionWindow) throws EXCEPTION {
+    var windowFunctions =
+        transformList(consistentPartitionWindow.getWindowFunctions(), this::visitWindowRelFunction);
+    var partitionExpressions =
+        transformList(
+            consistentPartitionWindow.getPartitionExpressions(),
+            t -> t.accept(getExpressionCopyOnWriteVisitor()));
+    var sorts = transformList(consistentPartitionWindow.getSorts(), this::visitSortField);
+
+    return Optional.of(
```

Comment:
To follow the pattern of the other copy on write visitors, this should return empty _unless_ any of the Optionals `windowFunctions`, `partitionExpressions` and `sorts` is non-empty.

Reply:
Good catch